src/PublicSale/lib.mo

```diff
@@ -270,6 +339,11 @@ module {
                         token;
                         memo;
                     });
+                    switch (allowlistAccount) {
+                        // Mint was successful, so we can remove allowlist entry.
+                        case (? aId) Allowlist.consumeAllowlist(aId, allowlist);
+                        case (_) {} // Not presale, nothing to do there.
+                    };
```

Comment:
Consuming the allow list in `lock` means that someone who runs into an error the the mint might "lose" their whitelist mint. Since we only allow one lock per person, moving this to `settle` may be a safe bet.